src/sentry/workflow_engine/endpoints/organization_workflow_details.py

```diff
@@ -0,0 +1,78 @@
+from drf_spectacular.utils import extend_schema
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import OrganizationEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.apidocs.constants import (
+    RESPONSE_BAD_REQUEST,
+    RESPONSE_FORBIDDEN,
+    RESPONSE_NOT_FOUND,
+    RESPONSE_UNAUTHORIZED,
+)
+from sentry.apidocs.parameters import GlobalParams, WorkflowParams
+from sentry.models.organization import Organization
+from sentry.workflow_engine.endpoints.serializers import WorkflowSerializer
+from sentry.workflow_engine.models import Workflow
+
+
+@region_silo_endpoint
+class OrganizationWorkflowDetailsEndpoint(OrganizationEndpoint):
+    def convert_args(self, request: Request, workflow_id, *args, **kwargs):
+        args, kwargs = super().convert_args(request, *args, **kwargs)
+        try:
+            kwargs["workflow"] = Workflow.objects.get(
+                organization=kwargs["organization"], id=workflow_id
+            )
+        except Workflow.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return args, kwargs
```

Comment:
when we have the WorkflowIndexEndpoint GET, will this be moved to a shared base class for the Details and Index endpoints?

Reply:
do index endpoints typically have a `convert_args`? We do have that endpoint [here](https://github.com/getsentry/sentry/blob/master/src/sentry/workflow_engine/endpoints/organization_workflow_index.py#L40)

Reply:
i think you're right 🤔 maybe i'm not used to seeing `convert_args` here, i think for DetailEndpoints we often pass the `id` and try to get the object from inside the API method (without `convert_args`)